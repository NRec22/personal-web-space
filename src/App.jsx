import React from 'react';

const App = () => <h1>Hello, world!</h1>;

const Hello = () => 'Hello, world!';

export { App, Hello };
