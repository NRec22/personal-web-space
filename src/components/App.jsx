import React, { Fragment } from 'react';

const App = () => (
  <Fragment>
    <h1>Neven Recchia</h1>
    <h2>Home</h2>
    <h2>About Me</h2>
    <h2>Contact</h2>
    <h2>LinkedIn</h2>
    <h2>Github</h2>
  </Fragment>
);

export default App;
