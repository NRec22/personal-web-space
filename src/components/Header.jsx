import React from 'react';
import Navbar from 'react-bulma-components/lib/components/navbar';

const Header = () => (
  <Navbar color="light">
    <Navbar.Brand>
      <Navbar.Item>...</Navbar.Item>
      <Navbar.Burger />
    </Navbar.Brand>
    <Navbar.Menu>
      <Navbar.Container>
        <Navbar.Item>Home</Navbar.Item>
        <Navbar.Item>About Me</Navbar.Item>
        <Navbar.Item>LinkedIn</Navbar.Item>
        <Navbar.Item>Github</Navbar.Item>
        <Navbar.Item>Contact Me</Navbar.Item>
      </Navbar.Container>
    </Navbar.Menu>
  </Navbar>
);

export default Header;