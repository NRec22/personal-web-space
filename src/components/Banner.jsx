import React from 'react';
import Hero from 'react-bulma-components/lib/components/hero';
import Container from 'react-bulma-components/lib/components/container';
import Heading from 'react-bulma-components/lib/components/heading';
import Header from './Header';

const Banner = () => (
  <Hero color="dark" size="medium" gradient>
    <Header />
    <Hero.Body>
      <Container>
        <Heading spaced size={1}>Neven Recchia</Heading>
        <Heading subtitle size={5}>Aspiring developer.</Heading>
      </Container>
    </Hero.Body>
  </Hero>
)

export default Banner;