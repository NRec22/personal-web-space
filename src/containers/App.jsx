import React, { Fragment } from 'react';
import Header from '../components/Header';
import Banner from '../components/Banner';

const App = () => (
  <Fragment>
    <Header />
    <Banner />
  </Fragment>
);

export default App;
