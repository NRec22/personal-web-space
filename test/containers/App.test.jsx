import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import App from '../../src/containers/App';
import Header from '../../src/components/Header';
import Banner from '../../src/components/Banner';

describe('App:', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  })
  it('should contain a header', () => {
    // const wrapper = shallow(<App />);
    expect(wrapper.find(Header)).toHaveLength(1);
  })

  it('should contain a banner', () => {
    expect(wrapper.find(Banner)).toHaveLength(1);
  })

  it('should render as expected', () => {
    // const wrapper = shallow(<App />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})
