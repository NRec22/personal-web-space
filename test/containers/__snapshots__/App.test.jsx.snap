// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: should render as expected 1`] = `
<React.Fragment>
  <Header />
  <Banner />
</React.Fragment>
`;
